pkg/dns/client/dns.go

```diff
@@ -187,11 +216,17 @@ func (h *LocalDNSServer) ServeDNS(proxy *dnsProxy, w dns.ResponseWriter, req *dn
 	lp := h.lookupTable.Load()
 	hostname := strings.ToLower(req.Question[0].Name)
 	if lp == nil {
-		log.Debugf("dns request for host %q before lookup table is loaded", hostname)
-		response = new(dns.Msg)
-		response.SetReply(req)
-		response.Rcode = dns.RcodeServerFailure
-		_ = w.WriteMsg(response)
+		if strings.HasSuffix(h.addr, ":53") {
```

Comment:
nit: would be good to add a comment why its safe to resolve before loaded here

Reply:
Also very nit: may prefer a function/struct field to store "directMode" (or better name). Or at least a comment that :53 == direct mode